### ISSUE TITLE
Useless loss function?

### ISSUE THREAD

**Issue by yardenoren** [carpedm20/DCGAN-tensorflow#211]
line 126 in model.py:

`sigmoid_cross_entropy_with_logits(self.D_logits_, tf.zeros_like(self.D_)))`

the cross entropy is defined as:

sum_i [ y_i * log(y'_i) ] 

so the the loss function is already at the minimum - 0.
am i missing something?

**Comment by yardal** [comment]
+1

**Comment by ppwwyyxx** [comment]
Google it: https://www.tensorflow.org/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits
```
For brevity, let x = logits, z = labels. The logistic loss is

  z * -log(sigmoid(x)) + (1 - z) * -log(1 - sigmoid(x))
= z * -log(1 / (1 + exp(-x))) + (1 - z) * -log(exp(-x) / (1 + exp(-x)))
= z * log(1 + exp(-x)) + (1 - z) * (-log(exp(-x)) + log(1 + exp(-x)))
= z * log(1 + exp(-x)) + (1 - z) * (x + log(1 + exp(-x))
= (1 - z) * x + log(1 + exp(-x))
= x - x * z + log(1 + exp(-x))
```
